src/trigger.h

```diff
@@ -7,7 +7,7 @@
 #include "chunk.h"
 
 #define trigger_is_chunk_trigger(trigger) \
-	((trigger) != NULL && TRIGGER_FOR_ROW((trigger)->tgtype) && !(trigger)->tgisinternal)
+	((trigger) != NULL && TRIGGER_FOR_ROW((trigger)->tgtype) && !(trigger)->tgisinternal && strncmp(trigger->tgname, INSERT_BLOCKER_NAME, strlen(INSERT_BLOCKER_NAME)))
```

Comment:
Should this be `NameStr(trigger->tgname )`? If we're using tests like this I'd be happier if we had an assertion _somewhere_ that `strlen(INSERT_BLOCKER_NAME) < NAMEDATALEN`, since postgres will happily truncate names without warning. Maybe at `insert_blocker_trigger_add`? (Since we do this all over the place it's not critical, and shouldn't block the  PR).

Reply:
Hm, this is odd. I actually had NameStr at somepoint and changed it because I saw somewhere that `tgname` was a text/string, but now I'm looking it up and it's clearly a `name`.... If it's a `name` I will definitely change, but just let me confirm.

Reply:
you should do `strcmp` here since you know that `INSERT_BLOCKER_NAME` is `\0` terminated. Basically, there is no reason to call `strncmp` with `strlen()` of compared name since the point of passing `n` is to avoid the `\0` char search. Further, passing in `strlen(INSERT_BLOCKER_NAME)` here means that you are actually comparing prefixes, which we don't want in this case since non-internal triggers do not have the OID appended to the name.

Reply:
Also, I think we should be explicit with what value we want here, e.g., `strcmp() != 0`, and you should do `(trigger)->tgname` rather than `trigger` in the macro to ensure passing in something like `trigger_is_chunk_trigger(ptr + offset)` works.